File: Ruby/kex.rb
require_relative 'player'
require_relative 'kex_1sub'

class Kex
  def execute
    puts '冒険が今始まる！'
    player = PlayerEx.new
    continue_game = 1
    kills = 0

    while continue_game != 0
      puts "現HP：#{player.hp}"
      print '奥に進みますか？（１：奥に進む　０．帰る）＞'
      continue_game = gets.chomp.to_i
      if continue_game != 0
        golem_lv = rand(0..4)
        attack_golem_ex(golem_lv, player)
        kills += 1
        break if player.hp.zero?
      end
    end
    puts 'リ〇ミト！'
    puts "戦闘回数：#{kills}　残りHP：#{player.hp}"
  end
end

File: Cpp/Kex.cpp
#include "Kex.h"

#include <iostream>
#include <random>

#include "Kex_1sub.h"

using std::cin;
using std::cout;
using std::mt19937;
using std::uniform_int_distribution;

void Kex::Execute() {
    auto engine = Kex_1sub::GetSeededEngine<>();
    uniform_int_distribution<> generator(0, 4);
    cout << "冒険が今始まる！\n";

    PlayerEx player;
    auto continue_game = true;
    auto kills = 0;

    do {
        cout << "\n現HP：" << player.Hp << "\n";
        cout << "奥に進みますか？（１：奥に進む　０．帰る）＞";
        auto choice = 0;
        cin >> choice;
        continue_game = static_cast<bool>(choice);

        if (continue_game) {
            auto golemLv = generator(engine);
            Kex_1sub::AttackGolem(golemLv, player);
            ++kills;
            if (player.Hp == 0) {
                break;
            }
        }
    } while (continue_game);

    cout << "リ〇ミト！\n\n";
    cout << "戦闘回数：" << kills << "　残りHP：" << player.Hp;
}